### PR TITLE
[CL-29] Modal Components for the component library

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The purpose of this PR is to add Modals into the component library.

To note, this PR is focused on the visual of the modal that will "pop-up" and not the actual pop-up implementation.

The two modal templates are as follows:

### Default Icon
```
<bit-simple-modal>
  <span bit-modal-title>Alert Modal Title</span>
  <span bit-modal-content>Message Content</span>
   <div bit-modal-footer class="tw-flex tw-flex-row tw-gap-2">
      <button bitButton buttonType="primary"> Yes </button>
      <button bitButton buttonType="secondary"> No </button>
   </div>
</bit-simple-modal>
```

### Custom Icon
```
<bit-simple-modal>
  <i bit-modal-icon class="bwi bwi-star tw-text-3xl tw-text-success" aria-hidden="true"></i>
  <span bit-modal-title>Premium Subscription Available</span>
  <span bit-modal-content>Message Content</span>
   <div bit-modal-footer class="tw-flex tw-flex-row tw-gap-2">
      <button bitButton buttonType="primary"> Yes </button>
      <button bitButton buttonType="secondary"> No </button>
   </div>
</bit-simple-modal>
```

The keywords bit-modal-icon, bit-modal-title, bit-modal-content, and bit-modal-footer determine placement of content in the modal using angular's multi-slot content project.
https://angular.io/guide/content-projection#multi-slot-content-projection

### Modal
```
 <bit-modal [modalSize]="modalSize">
    <span bit-modal-title> Modal Title </span>
    <span bit-modal-content>
    Modal body text goes here.
    </span>
    <div bit-modal-footer class="tw-flex tw-flex-row tw-gap-2">
      <button bitButton buttonType="primary"> Save </button>
      <button bitButton buttonType="secondary"> Cancel </button>
    </div>
  </bit-modal>
```
Where "modalSize" can be small, default, or large, controlling the width of the modal displayed.

The keywords bit-modal-title, bit-modal-content, and bit-modal-footer determine placement of content in the modal using angular's multi-slot content project.
https://angular.io/guide/content-projection#multi-slot-content-projection

## Code changes

Code for the simple modal template.

libs/components/src/modal/modal-simple.component.html
libs/components/src/modal/modal-simple.component.ts
libs/components/src/modal/modal-simple.stories.ts

Code for the modal template.
libs/components/src/modal/modal.component.html
libs/components/src/modal/modal.component.ts
libs/components/src/modal/modal.stories.ts

The Modals components can be viewed by running `npm run storybook` at the root of the project.

### Simple Modal
image

image

### Modal

image

image

image

## Before you submit

```
- [X] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

**Review by Hinton — COMMENTED** [review]
Some general feedback.

1. We should avoid non standard size. Check with design which [sizes](https://tailwindcss.com/docs/max-width) they would like to use. We should probably also use `max-width` instead of `width` to be a bit responsive.
2. Try and use a single width per modal size, the body and title widths can be generated using margins and/or percentages.

As to the height, https://tailwindcss.com/docs/max-height combined with padding should work? Might need to put the max-height on a parent container, and use height 100% on the inner though.

**Comment by Hinton** [comment]
@danielleflinn for the simple modals, do we want the developers to use any icons or just a set number, similar to callouts?

**Comment by danielleflinn** [comment]
@Hinton lets default the component to the bwi-exclamation-triangle, but allow it to be configurable to any icon per implementation.

**Comment by Thomas-Avery** [comment]
Visually, everything should now be implemented and is ready for review. 

- fixed height scrolling to work as requested
- responsive width now implemented
- added the "X" close icon to Modals
- moved all examples to using bitButtons 
- added example stories to show content expansion and scrolling for both Simple Modal and Modal

Here is the proposal for implementing the default bwi-exclamation-triangle icon. 
This proposal is now implemented as well.

## Simple Modal

### Default Icon
```
<bit-simple-modal [useDefaultIcon]="true">
  <span title>Alert Modal Title</span>
  <span message>Message Content</span>
  <button footer bitButton buttonType="primary">Yes</button>
  <button footer bitButton buttonType="secondary">No</button>
</bit-simple-modal>
```

### Custom Icon
```
<bit-simple-modal [useDefaultIcon]="false">
  <i icon class="bwi bwi-star tw-text-3xl tw-text-success" aria-hidden="true"></i>
  <span title>Premium Subscription Available</span>
  <span message>Message Content</span>
  <button footer bitButton buttonType="primary">Yes</button>
  <button footer bitButton buttonType="secondary">No</button>
</bit-simple-modal>
```

Instead of using the keyword vector, I think "icon" works better for the template.

File: libs/components/src/modal/modal-simple.stories.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+const TemplateScroll: Story<ModalSimpleComponent> = (args: ModalSimpleComponent) => ({
+  props: args,
+  template: `
+  <bit-simple-modal [useDefaultIcon]="useDefaultIcon">
+      <span title> Alert Modal Title
+      </span>
+      <span message> Message Content
+      Message text goes here.<br>
+      repeating lines of characters
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.stories.ts:58]: Lets make this an `ng-for`, so we can actually follow what happens here.

File: libs/components/src/modal/modal-simple.component.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+  selector: "bit-simple-modal",
+  templateUrl: "./modal-simple.component.html",
+})
+export class ModalSimpleComponent {
+  @Input()
+  useDefaultIcon: boolean;
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.component.ts:9]: We should be able to use `ViewChild` here to determine if the icon exists.

**Review by Hinton — COMMENTED** [review]
Some initial review comments while I review the rest.

File: libs/components/src/modal/modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+  selector: "bit-modal",
+  templateUrl: "./modal.component.html",
+})
+export class ModalComponent {
+  // Controls the size of the bit modal displayed. Needs to be implemented.
+  @Input()
+  modalSize: "small" | "default" | "large";
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.ts:10]: Lets keep these on the same line.

```suggestion
  @Input() modalSize: "small" | "default" | "large";
```

File: libs/components/src/modal/modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  getWidth() {
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.ts:12]: Let's change this to a getter.

File: libs/components/src/modal/modal.component.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  >
+    <!-- Modal Footer Ruler-->
+    <div
+      class="tw-w-full tw-box-border tw-h-px tw-bg-secondary-300 tw-border-solid tw-border-secondary-300 tw-border-0 tw-border-t tw-flex-none tw-order-none tw-self-stretch tw-grow-0"
+    ></div>
+    <!-- Modal Footer Buttons-->
+    <div
+      class="tw-w-full tw-flex tw-flex-row tw-items-center tw-p-4 tw-gap-2 tw-max-h-full tw-flex-none tw-order-1 tw-self-stretch tw-grow-0"
+    >
+      <ng-content select="[footer]"></ng-content>
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:48]: Let's use the `bit` prefix on these selectors.

File: libs/components/src/modal/modal.stories.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  component: ModalComponent,
+  decorators: [
+    moduleMetadata({
+      imports: [ButtonModule],
+    }),
+  ],
+  args: {
+    modalSize: "small",
+  },
+} as Meta;
```
Comment (Hinton) [inline libs/components/src/modal/modal.stories.ts:18]: Can we add the design link here?

File: libs/components/src/modal/modal.component.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    class="tw-w-full tw-flex tw-flex-col tw-justify-end tw-items-end tw-p-0 tw-max-h-full tw-bg-background-alt tw-flex-none tw-order-2 tw-self-stretch tw-grow-0"
+  >
+    <!-- Modal Footer Ruler-->
+    <div
+      class="tw-w-full tw-box-border tw-h-px tw-bg-secondary-300 tw-border-solid tw-border-secondary-300 tw-border-0 tw-border-t tw-flex-none tw-order-none tw-self-stretch tw-grow-0"
+    ></div>
+    <!-- Modal Footer Buttons-->
+    <div
+      class="tw-w-full tw-flex tw-flex-row tw-items-center tw-p-4 tw-gap-2 tw-max-h-full tw-flex-none tw-order-1 tw-self-stretch tw-grow-0"
+    >
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:47]: Why do we need 3 classes for the footer? Could we not use a single div for everything?

File: libs/components/src/modal/modal.component.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  <div
+    class="tw-w-full tw-flex tw-flex-col tw-justify-center tw-items-center tw-order-1 tw-self-stretch tw-grow-0 tw-relative tw-flex-auto tw-overflow-y-auto"
+  >
+    <!-- Modal Body Text-->
+    <div
+      class="tw-w-full tw-pb-8 tw-p-4 tw-not-italic tw-font-normal tw-text-base tw-text-main tw-order-none tw-self-stretch tw-grow-0 tw-relative tw-flex-auto tw-overflow-y-auto"
+    >
+      <ng-content select="[body]"></ng-content>
+    </div>
+  </div>
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:34]: Do we need two divs here?

File: libs/components/src/modal/modal.component.html [hunk 9]
```diff
@@ -0,0 +1,51 @@
+<!-- Popup shell -->
+<div
+  [ngClass]="getWidth()"
+  class="tw-max-h-screen tw-my-4 tw-box-border tw-flex tw-flex-col tw-items-start tw-p-0 tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-flex-none tw-order-0 tw-grow-1"
+>
+  <!-- Modal Header-->
+  <div
+    class="tw-w-full tw-box-border tw-flex tw-flex-row tw-items-center tw-gap-4 tw-p-4 tw-max-h-full tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-flex-none tw-order-none tw-self-stretch tw-grow-0"
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:8]: Do you need `order-none`?

**Review by Hinton — DISMISSED** [review]

File: libs/components/src/modal/modal.component.html (hunk 5 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html:48]: Should we go with:

bit-modal-title
bit-modal-body
bit-modal-footer

or 

bit-title
bit-body
bit-footer

The reason I bring it up is if there will ever be any future conflicts with other bit-<something> keywords.

File: libs/components/src/modal/modal.component.html (hunk 7 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html:47]: We can merge the footer ruler into the main footer div. 

The modal footer buttons div provides the padding and gap to center the buttons inside the main footer div. 

image

Unless I'm missing something, are you wanting consumers to implement their own padding and gap styling?

I just was going off the figma models, which have 3 separate containers.

File: libs/components/src/modal/modal.component.html (hunk 8 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html:34]: Same as the footer, the body text div provides padding to center inside the main body container.

File: libs/components/src/modal/modal-simple.stories.ts (hunk 1 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.stories.ts]: solved in https://github.com/bitwarden/clients/pull/3053/commits/dd115078508a8bf1c648d415eeb0baafbd2ceab8

File: libs/components/src/modal/modal.component.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+  selector: "bit-modal",
+  templateUrl: "./modal.component.html",
+})
+export class ModalComponent {
+  // Controls the size of the bit modal displayed. Needs to be implemented.
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.ts]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/dd115078508a8bf1c648d415eeb0baafbd2ceab8

File: libs/components/src/modal/modal.component.ts (hunk 4 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.ts]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/dd115078508a8bf1c648d415eeb0baafbd2ceab8

File: libs/components/src/modal/modal.stories.ts (hunk 6 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.stories.ts]: resovled in https://github.com/bitwarden/clients/pull/3053/commits/dd115078508a8bf1c648d415eeb0baafbd2ceab8

File: libs/components/src/modal/modal.component.html (hunk 7 above)
Comment (Hinton) [reply libs/components/src/modal/modal.component.html]: I'm not really following. What's stopping us from having padding on the outer div?

Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/c4d09a5eaa0ff6d0cfc5a943fe8938a48a03b039

File: libs/components/src/modal/modal.component.html (hunk 8 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/c4d09a5eaa0ff6d0cfc5a943fe8938a48a03b039

File: libs/components/src/modal/modal.component.html (hunk 9 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/c4d09a5eaa0ff6d0cfc5a943fe8938a48a03b039

File: libs/components/src/modal/modal.component.html [hunk 11]
```diff
@@ -0,0 +1,51 @@
+<!-- Popup shell -->
+<div
+  [ngClass]="getWidth()"
+  class="tw-max-h-screen tw-my-4 tw-box-border tw-flex tw-flex-col tw-items-start tw-p-0 tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-flex-none tw-order-0 tw-grow-1"
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/a8ea95db1f9b8c0a4e729c3893e6740a627a47b0

File: libs/components/src/modal/modal-simple.component.ts (hunk 2 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.component.ts]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/27fbba52ff92a089fbd15c77a43ee7757c434671

File: libs/components/src/modal/modal.component.html [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  <!-- Modal Body-->
+  <div
+    class="tw-pb-8 tw-p-4 tw-w-full tw-flex tw-flex-col tw-order-1 tw-self-stretch tw-grow-0 tw-relative tw-flex-auto tw-overflow-y-auto tw-not-italic tw-font-normal tw-text-base tw-text-main"
+  >
+    <ng-content select="[bit-modal-body]"></ng-content>
+  </div>
+
+  <!-- Modal Footer-->
+  <div
+    class="tw-w-full tw-flex tw-flex-row tw-items-center tw-p-4 tw-gap-2 tw-max-h-full tw-bg-background-alt tw-flex-none tw-order-2 tw-self-stretch tw-grow-0 tw-border-solid tw-border-secondary-300 tw-border-0 tw-border-t"
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:31]:
```suggestion
    class="tw-p-4 tw-bg-background-alt tw-border-solid tw-border-secondary-300 tw-border-0 tw-border-t"
```

File: libs/components/src/modal/modal.component.html [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    <!-- Modal Title close button container TODO: when implementing modal popup this should broadcast to close modal popup-->
+    <button class="tw-border-0 tw-bg-transparent tw-text-main tw-p-0">
+      <i class="bwi bwi-close tw-text-xs tw-font-bold tw-text-main" aria-hidden="true"></i>
+    </button>
+  </div>
+
+  <!-- Modal Body-->
+  <div
+    class="tw-pb-8 tw-p-4 tw-w-full tw-flex tw-flex-col tw-order-1 tw-self-stretch tw-grow-0 tw-relative tw-flex-auto tw-overflow-y-auto tw-not-italic tw-font-normal tw-text-base tw-text-main"
+  >
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:25]:
```suggestion
  <div class="tw-pb-8 tw-p-4 tw-w-full tw-relative tw-overflow-y-auto">
```

File: libs/components/src/modal/modal.component.html [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    class="tw-w-full tw-flex tw-flex-row tw-items-center tw-gap-4 tw-p-4 tw-max-h-full tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-flex-none tw-self-stretch tw-grow-0"
+  >
+    <!-- Modal Title text container -->
+    <h2
+      class="tw-text-lg tw-font-normal tw-leading-7 tw-uppercase tw-flex tw-items-center tw-text-main tw-flex-none tw-order-none tw-grow tw-mb-0"
+    >
+      <ng-content select="[bit-modal-title]"></ng-content>
+    </h2>
+    <!-- Modal Title close button container TODO: when implementing modal popup this should broadcast to close modal popup-->
+    <button class="tw-border-0 tw-bg-transparent tw-text-main tw-p-0">
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:17]:
```suggestion
    <button class="tw-border-0 tw-bg-transparent tw-p-0">
```

File: libs/components/src/modal/modal.component.html [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      <i class="bwi bwi-close tw-text-xs tw-font-bold tw-text-main" aria-hidden="true"></i>
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:18]:
```suggestion
      <i class="bwi bwi-close tw-text-xs tw-font-bold" aria-hidden="true"></i>
```

**Review by Hinton — COMMENTED** [review]
There are quite a few unnecessary css classes we can remove, I've posted my suggestions for the regular modal. Please look into doing the same for the simple modal, as the fewer classes we use the easier it is to maintain.

We should also remove the html comments unless they bring value, since we've eliminated most of the unnecessary divs I don't view the comments as providing much value any longer as the selector already describes what's in the divs.

File: libs/components/src/modal/modal.component.html [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+  [ngClass]="width"
+  class="tw-max-h-screen tw-my-4 tw-flex tw-flex-col tw-items-start tw-p-0 tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-overflow-hidden"
+>
+  <!-- Modal Header-->
+  <div
+    class="tw-w-full tw-flex tw-flex-row tw-items-center tw-gap-4 tw-p-4 tw-max-h-full tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-flex-none tw-self-stretch tw-grow-0"
+  >
+    <!-- Modal Title text container -->
+    <h2
+      class="tw-text-lg tw-font-normal tw-leading-7 tw-uppercase tw-flex tw-items-center tw-text-main tw-flex-none tw-order-none tw-grow tw-mb-0"
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

File: libs/components/src/modal/modal.component.html [hunk 17]
```diff
@@ -0,0 +1,35 @@
+<!-- Popup shell -->
+<div
+  [ngClass]="width"
+  class="tw-max-h-screen tw-my-4 tw-flex tw-flex-col tw-items-start tw-p-0 tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-overflow-hidden"
+>
+  <!-- Modal Header-->
+  <div
+    class="tw-w-full tw-flex tw-flex-row tw-items-center tw-gap-4 tw-p-4 tw-max-h-full tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-flex-none tw-self-stretch tw-grow-0"
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: Resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

File: libs/components/src/modal/modal.component.html [hunk 18]
```diff
@@ -0,0 +1,35 @@
+<!-- Popup shell -->
+<div
+  [ngClass]="width"
+  class="tw-max-h-screen tw-my-4 tw-flex tw-flex-col tw-items-start tw-p-0 tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-overflow-hidden"
```
[the remark this answers is not in the document]
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

did not take out tw-overflow-hidden as discussed.

File: libs/components/src/modal/modal.component.html (hunk 12 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

File: libs/components/src/modal/modal.component.html (hunk 13 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

File: libs/components/src/modal/modal.component.html (hunk 14 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

File: libs/components/src/modal/modal.component.html (hunk 15 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html:12]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/9f937eaff562bd6e7f050066a3448336049c9436

Comment (Hinton) [reply libs/components/src/modal/modal.component.html:12]: Hmm that's true, bootstrap styling messing things up again.

File: libs/components/src/modal/modal.component.html [hunk 19]
```diff
@@ -0,0 +1,25 @@
+<div
+  [ngClass]="width"
+  class="tw-max-h-screen tw-my-4 tw-flex tw-flex-col tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-text-main tw-overflow-hidden"
+>
+  <div
+    class="tw-flex tw-gap-4 tw-p-4 tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300 tw-text-main"
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:6]: Color gets inherited from the parent div.
```suggestion
    class="tw-flex tw-gap-4 tw-p-4 tw-border-0 tw-border-b tw-border-solid tw-border-secondary-300"
```

File: libs/components/src/modal/modal.component.html [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+  >
+    <h2 class="tw-text-lg tw-uppercase tw-flex tw-grow tw-mb-0">
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:8]: Is display flex needed here?
```suggestion
    <h2 class="tw-text-lg tw-uppercase tw-grow tw-mb-0">
```

File: libs/components/src/modal/modal.component.html [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+      <ng-content select="[bit-modal-title]"></ng-content>
+    </h2>
+    <button class="tw-border-0 tw-bg-transparent tw-p-0">
+      <i class="bwi bwi-close tw-text-xs tw-font-bold tw-text-main" aria-hidden="true"></i>
+    </button>
+  </div>
+
+  <div class="tw-pb-8 tw-p-4 tw-w-full tw-relative tw-overflow-y-auto">
```
Comment (Hinton) [inline libs/components/src/modal/modal.component.html:16]: Is Width and relative needed here?

```suggestion
  <div class="tw-pb-8 tw-p-4 tw-overflow-y-auto">
```

File: libs/components/src/modal/modal-simple.component.html [hunk 22]
```diff
@@ -0,0 +1,23 @@
+<div
+  class="tw-max-w-sm tw-max-h-screen tw-my-4 tw-flex tw-flex-col tw-bg-text-contrast tw-border tw-border-solid tw-border-secondary-300 tw-rounded tw-text-main tw-overflow-hidden"
+>
+  <div
+    class="tw-text-center tw-text-main tw-flex tw-flex-col tw-items-center tw-px-4 tw-pt-4 tw-gap-2"
+  >
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.component.html:6]:
```suggestion
  <div class="tw-text-center tw-flex tw-flex-col tw-items-center tw-px-4 tw-pt-4 tw-gap-2">
```

File: libs/components/src/modal/modal-simple.component.html [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+    <ng-content *ngIf="hasIcon; else elseBlock" select="[bit-sm-icon]"></ng-content>
+    <ng-template #elseBlock>
+      <i class="bwi bwi-exclamation-triangle tw-text-3xl tw-text-warning" aria-hidden="true"></i>
+    </ng-template>
+    <h2 class="tw-font-semibold tw-text-base tw-flex-none tw-mb-0">
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.component.html:11]:
```suggestion
    <h2 class="tw-font-semibold tw-text-base tw-mb-0">
```

File: libs/components/src/modal/modal-simple.component.html [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+      <ng-content select="[bit-sm-title]"></ng-content>
+    </h2>
+  </div>
+  <div
+    class="tw-text-center tw-text-main tw-text-base tw-pt-2 tw-pb-4 tw-px-4 tw-relative tw-overflow-y-auto"
+  >
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.component.html:17]:
```suggestion
  <div class="tw-text-center tw-text-base tw-pt-2 tw-pb-4 tw-px-4 tw-overflow-y-auto">
```

File: libs/components/src/modal/modal-simple.stories.ts [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+  `,
+});
+
+export const CustomIcon = TemplateWithIcon.bind({});
+
+const TemplateScroll: Story<ModalSimpleComponent> = (args: ModalSimpleComponent) => ({
+  props: args,
+  template: `
+  <bit-simple-modal>
+      <span bit-sm-title> Alert Modal
```
Comment (Hinton) [inline libs/components/src/modal/modal-simple.stories.ts:65]: Lets use the same name for these across both components, so `bit-modal-title`.

Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.stories.ts]: Agreed: 

These names I think fit both well:

`bit-modal-title`
`bit-modal-content`
`bit-modal-footer`

and for the simple modal as well `bit-modal-icon`

File: libs/components/src/modal/modal-simple.component.html (hunk 24 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

File: libs/components/src/modal/modal-simple.component.html (hunk 23 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

File: libs/components/src/modal/modal-simple.component.html (hunk 22 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal-simple.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

File: libs/components/src/modal/modal.component.html (hunk 21 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

File: libs/components/src/modal/modal.component.html (hunk 20 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

File: libs/components/src/modal/modal.component.html (hunk 19 above)
Comment (Thomas-Avery) [reply libs/components/src/modal/modal.component.html]: resolved in https://github.com/bitwarden/clients/pull/3053/commits/938f2362c707729d900ffabad29b2289efe7fd5a

**Review by Hinton — COMMENTED** [review]
I think everything looks good from a code perspective. @danielleflinn do you mind reviewing the design before we merge this?

**Pull request merged**